components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -314,6 +315,17 @@ bool ResetOnFileTaskRunner(const base::FilePath& path) {
   return base::DeleteFile(path, false);
 }
 
+bool ResetOnFilesTaskRunner(const std::vector<base::FilePath>& paths) {
```

Comment:
the whole reset functionality was added for already unrecovered conditions, when user has a broken state already. How could it be more risky? I agree it can be done gracefully, but it is already risky, nothing to loose anymore.

Reply:
Please use `ResetAllState` and see `ShutdownBatAds` for safe way to shutdown

Reply:
We could be saving to the file as it is being deleted. See `ResetAllState` comments for a safer solution for ads

Reply:
NONCRITICAL
Covered by: brave/brave-browser#6239